src/descriptor/template.rs

```diff
@@ -680,6 +845,8 @@ mod test {
             Bip49(prvkey, KeychainKind::External).build(Network::Bitcoin),
             true,
             false,
+            false,
+            Network::Regtest,
```

Comment:
I was a bit confused with this at first. Why a `Mainnet` descriptor matches with a `Regtest` address? That's because the first `network` is used for setting the 2nd derivation index of `Bipxx`, and the second network is used for the address prefix.

As a result, if someone actually tries with the same `Xpriv` but builds with `build(Network::Regtest)` and derives the addresses, they will not match up with the test vector. 

So does it make sense to change the first network into `regtest` in the build call, or change the second network to `mainnet`? The first option is a much smaller changeset and can be done in a separate PR.

Reply:
@rajarshimaitra , you are right this is a bit confusing. I will opt for your first option. I will option a small PR for it before end of the week